src/include/cache.i

```diff
@@ -68,6 +79,21 @@ __wt_cache_read_gen_new(WT_SESSION_IMPL *session, WT_PAGE *page)
 }
 
 /*
+ * __wt_cache_stuck --
+ *      Indicate if the cache is stuck (i.e., not making progress).
+ */
+static inline bool
+__wt_cache_stuck(WT_SESSION_IMPL *session)
+{
+	WT_CACHE *cache;
+
+	cache = S2C(session)->cache;
+	return (cache->evict_aggressive_score == WT_EVICT_SCORE_MAX &&
```

Comment:
I generally like the new aggressive score scheme. I'm a little bit nervous that it is prone to off-by-ones and potentially wrapping. The usages look correct at the moment, but I'm inclined to abstract out all updates to ``cache->evict_aggressive_score`` eventually to make it less likely that we will introduce bugs in the future.